manifests/config.pp

```diff
@@ -110,6 +110,39 @@
     }
   }
 
+  # TODO: promtool currently does not support checking the syntax of file_sd_config "includes".
+  # Ideally we'd check them the same way the other config files are checked.
+  file { "${prometheus::config_dir}/file_sd_config.d":
+    ensure  => directory,
+    owner   => $prometheus::user,
+    group   => $prometheus::group,
+    purge   => true,
+    recurse => true,
+  }
+
+  $prometheus::server::collect_scrape_jobs.each |Hash $job_definition| {
+    if !has_key($job_definition, 'job_name') {
+      fail('collected scrape job has no job_name!')
+    }
+
+    $job_name = $job_definition['job_name']
+
+    Prometheus::Scrape_job <<| job_name == $job_name |>> {
+      collect_dir => "${prometheus::config_dir}/file_sd_config.d",
+      notify      => Class['::prometheus::service_reload'],
+    }
+  }
+  # collect the scrape jobs in a single list that gets appended to
```

Comment:
nitpicking: "collect" here is a bit ambiguous, because it is also used to refer to puppet's export/collect mechanic. Maybe rephrase as "join the scape jobs (...)"?

Reply:
i used "assemble"